types/types.go

```diff
@@ -56,13 +58,33 @@ func (h *Hash) Scan(src interface{}) error {
 	return nil
 }
 
-func (a Address) EIP55() string {
-	// TODO
-	return hex.EncodeToHex(a[:])
+// eip55 returns the checksummed address with 0x prefix
+// https://github.com/ethereum/EIPs/blob/master/EIPS/eip-55.md
+func (a Address) eip55() string {
+	addrBytes := a.Bytes() // 20 bytes
+
+	// Encode to hex without the 0x prefix
+	lowercaseHex := hex.EncodeToHex(addrBytes)[2:]
+	hashedAddress := hex.EncodeToHex(keccak.Keccak256(nil, []byte(lowercaseHex)))[2:]
+
+	result := make([]rune, len(lowercaseHex))
+	// Iterate over each character in the hashed address
+	for idx, ch := range lowercaseHex {
```

Comment:
either the comment or code is wrong. It says `iterate over **hashed** address` but does `range lowercaseHex`?

Reply:
Comments lie 🥸

Fixed 👍